tests/unittest_inference.py

```diff
@@ -1703,7 +1703,8 @@ def __init__(self):
         """
         ast = extract_node(code, __name__)
         expr = ast.func.expr
-        self.assertIs(next(expr.infer()), util.Uninferable)
+        with pytest.raises(exceptions.InferenceError):
+            next(expr.infer())
```

Comment:
This test has changed behaviour. I assume that it's okay for it to raise `InferenceError` instead of returning `Uninferable` as long as it still doesn't leak `StopIteration`

Reply:
@cdce8p what is your opinion?

Reply:
I'm not too familiar with that part of the code base. If the pylint tests continue to pass, I would guess it's fine.

Reply:
Potentially this could cause new crash if an inference error is raised in calling code ? I don't think pylint tests cover all possible edge case.